src/Engine/Shaders/Shader.hpp

```diff
@@ -36,7 +39,7 @@ class Shader {
         void unloadTexture(int);
 
         /* Render functions */
-        virtual void render(const std::vector<Component *> & components) = 0;
+        virtual void render(const CameraComponent & camera, const std::vector<Component *> & components) = 0;
```

Comment:
What if a shader doesn't need a camera component?

Reply:
Changed to a pointer so if the scene doesn't have a camera, it can pass null. Basically, a shader shouldn't have a camera as part of its state. When its time to render, its given a camera from whose point to render if needed.

Reply:
What if it's not needed at all? Why should we force all shaders to even take a reference to a camera?